### PR TITLE
Define Transmission class, and also return promise

### PR DESCRIPTION
**Pull request by G-Ray**
The api should now be the same as https://github.com/G-Ray/transmission-native/

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,6 +46,18 @@ tr.request(
...
+    method: 'session-get',
+    arguments: {
+      fields: ['version'],
+    },
+  }
+)
+
+console.log(res)
+
 tr.close()
```
Comment (tony-go) [inline README.md:61]: Maybe we should start the doc with the Promise example? as it's the defacto standard.

File: src/index.tsx [hunk 2]
```diff
@@ -17,16 +17,30 @@ const Transmission = NativeModules.Transmission
...
-  );
-}
+  close() {
+    return Transmission.close();
+  }
+
+  request(req: any, cb?: any) {
+    // Callback api
+    if (cb && typeof cb === 'function') {
+      return Transmission.request(JSON.stringify(req), (err: any, res: any) => {
```
Comment (tony-go) [inline src/index.tsx:32]: We should definitely make `Transmission.request` return a promise.

Comment (tony-go) [reply src/index.tsx:32]: I mean in the future.

**Review by tony-go — APPROVED** [review]
lgtm % the doc comment

File: README.md (hunk 1 above)
Comment (G-Ray) [reply README.md:61]: Yep we sould open a PR for `react-native-transmission` & `transmission-native`

**Pull request merged**